cexec/main.go

```diff
@@ -1,111 +1,202 @@
 package main
 
 import (
+	"context"
+	"errors"
+	"flag"
 	"fmt"
+	"log/slog"
 	"os"
 	"os/exec"
+	"os/signal"
 	"path/filepath"
 	"strings"
 	"syscall"
 
-	log "github.com/sirupsen/logrus"
+	"github.com/peterbourgon/ff/v3"
 )
 
-const mountAction = "/mountAction"
+const (
+	mountAction = "/mountAction"
+	envPrefix   = "CEXEC"
+)
+
+type settings struct {
+	blockDevice        string
+	filesystemType     string
+	chroot             string
+	defaultInterpreter string
+	cmdLine            string
+	updateResolvConf   bool
+}
 
 func main() {
-	fmt.Printf("CEXEC - Chroot Exec\n------------------------\n")
+	ctx, done := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGHUP, syscall.SIGTERM)
+	defer done()
 
-	// Parse the environment variables that are passed into the action
-	blockDevice := os.Getenv("BLOCK_DEVICE")
-	filesystemType := os.Getenv("FS_TYPE")
-	chroot := os.Getenv("CHROOT")
-	defaultInterpreter := os.Getenv("DEFAULT_INTERPRETER")
-	cmdLine := os.Getenv("CMD_LINE")
+	logger := slog.New(slog.NewTextHandler(os.Stdout, nil))
+	fs := flag.NewFlagSet("cexec", flag.ExitOnError)
+	s := settings{}
+	fs.StringVar(&s.blockDevice, "block-device", "", "block device to mount (required)")
+	fs.StringVar(&s.filesystemType, "fs-type", "", "filesystem type (required)")
+	fs.StringVar(&s.chroot, "chroot", "", "use chroot environment to run given command (deprecated)")
+	fs.StringVar(&s.defaultInterpreter, "default-interpreter", "", "default interpreter (optional)")
+	fs.StringVar(&s.cmdLine, "cmd-line", "", "command line to execute (required)")
+	fs.BoolVar(&s.updateResolvConf, "update-resolv-conf", false, "update /etc/resolv.conf in chroot environment (optional)")
+	jsonLogger := fs.Bool("json-outout", true, "enable json output for logging")
 
-	var exitChroot func() error
+	if err := ff.Parse(fs, os.Args[1:], ff.WithEnvVarNoPrefix()); err != nil {
+		logger.Error(err.Error())
+		os.Exit(1)
+	}
+
+	// check for required fields to be set.
+	if missingFields := s.checkRequiredFields(); len(missingFields) > 0 {
+		fmt.Fprintln(os.Stderr, "missing required fields", missingFields)
+		fmt.Fprintln(os.Stderr)
+		fs.Usage()
+		os.Exit(10)
```

Comment:
Do we want to document exit code 10 in the README given we've specified it in this case?

Reply:
sure thing.